File: Python/problem1108.py
# -*- coding: utf-8 -*-
"""
Created on Wed Jul 10 09:47:39 2019

@author: ASUS
"""

class Solution:
    def defangIPaddr(self, address: str) -> str:
#        while '.' in address:
#            address.replace('.','[.]')
#        return adress
        
#        return address.replace('.','[.]')
        
        stack = []
        for k in address:
            if k != '.':
                stack.append(k)
            else:
                stack.append('[')
                stack.append(k)
                stack.append(']')
        return ''.join(stack)
        
solu = Solution()
address = "1.1.1.1"
#address = "255.100.50.0"
print(solu.defangIPaddr(address))

File: C++/problem1108.cpp
#include <iostream>
#include <iomanip>
#include <vector>
#include <map>
#include <set>
#include <algorithm>
#include <cmath>
#include <ctime>
#include <cstdlib>
#include <windows.h>

using namespace std;

class Solution {
public:
    string defangIPaddr(string address) {
        string stack;
        for (int k = 0; k < address.size(); ++k)
        {
            if (address[k] == '.')
            {
                stack.push_back('[');
                stack.push_back('.');
                stack.push_back(']');
            }
            else
            {
                stack.push_back(address[k]);
            }
        }
        return stack;
    }
};

int main()
{
    string address = "255.100.50.0";
    Solution solu;
    string ans = solu.defangIPaddr(address);
    cout << ans << endl;
    return 0;
}